### DISCUSSION TITLE
DB row is not exist when fastapi returned 201.

### DISCUSSION THREAD

**Discussion by rumbarum** [fastapi/fastapi#11407] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# No code
```

### Description

I am not sure this is belonging here, but I need help.

I have infra like below,
aws ALB -> EC2( Nginx(docker) -> FastAPI(docker) -> Sqlalchemy )-> aws RDS (Postgresql)

My QA engineer found data is not exist when POST request succeed.

So I inspected logs, nginx logs saying,
```10.0.15.3 - - [03/Apr/2024:09:18:29 +0000] "POST servcice_url HTTP/1.1" 201 269 "service_domain" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.0.0 Whale/3.25.232.19 Safari/537.36"```
But no data exist matching nginx log.
No errors were found on fastapi logs and DB logs.

I do not know where to start, 
If anyone has advice, thanks you in advance. 

### Operating System

Linux

### Operating System Details

aws linux 2

### FastAPI Version

0.103.2

### Pydantic Version

1.10.13

### Python Version

3.10.3

**Comment by acidjunk** [comment]
Can you share the relevant code/example?

**Reply by rumbarum** [reply]
@acidjunk 
Thanks for your interest.
Request went through layers, middle wares, router, service, repository and save on db with Sqlalchemy async session.
Not a tricky nor complex logic exists, just creating a row. 
I inspected logs, but no errors on Fastapi and RDS.
I am suspecting DB layer, as Fastapi is not resposnible for DB job.

**Reply by pythonweb2** [reply]
It is impossible to know just based on the information you have given why this would occur. You would need to share the business logic.

**Comment by YuriiMotov** [comment]
Let's close this for now to keep the discussions clean and organized.
If you’ve found a solution, please share it here.
If you still need help, feel free to let us know.
